sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -75,7 +75,12 @@ private[sql] case class InsertIntoHadoopFsRelationCommand(
         case (x, ys) if ys.length > 1 => "\"" + x + "\""
       }.mkString(", ")
       throw new AnalysisException(s"Duplicate column(s) : $duplicateColumns found, " +
-          s"cannot save to file.")
+        "cannot save to file.")
+    }
+
+    if (bucketSpec.exists(_.numBuckets <= 0)) {
```

Comment:
is this the only place we need to check? what if we put this check inside the constructor of `BucketSpec`?